### PR TITLE
Bugfix: Find Default Branch

### PR DESCRIPTION
**Pull request by brolewis**
The code assumed that `master` was the default branch name, when that is
not guaranteed to be true, especially as more hosts are moving to making
`main` the default branch name.

### CODE REVIEW

File: jgt_tools/check_version.py [hunk 1]
```diff
@@ -11,14 +11,32 @@ def _version_changed(pyproject_diff):
...
+    """Find the default branch for a git repository."""
+    remotes = subprocess.check_output(
+        ["git", "remote"], universal_newlines=True
+    ).splitlines()
+    if len(remotes) == 1:
+        remote = remotes[0]
+    elif "origin" in remotes:
+        remote = "origin"
+    else:
+        raise KeyError("Cannot determine remote")
```
Comment (dgou) [inline jgt_tools/check_version.py:24]: NB: If you had a separate "find remote" helper the code could be slightly simpler:
```
def find_remote():
    remotes = ...
    if len(remotes) == 1:
        return remotes[0]
    if "origin" in remotes:
        return "origin"
    raise KeyError(...)
```

**Review by dgou — APPROVED** [review]
LGTM. Minor NB suggestion.

**Review by nrccua-bradb — APPROVED** [review]

**Review by bradsbrown — APPROVED** [review]
don't know who that `nrccua-bradb` guy is, but *I* approve this PR, too

**Review by dgou — APPROVED** [review]
LGTM.

**Pull request merged**